x/dogfood/keeper/impl_sdk.go

```diff
@@ -165,48 +165,34 @@ func (k Keeper) ApplyAndReturnValidatorSetUpdates(
 }
 
 // IterateBondedValidatorsByPower is an implementation of the staking interface expected by
-// the SDK's gov module. It is used to iterate through the validators by power. We do not
-// implement this function intentionally, since our model of governance is not designed yet.
-// Instead of staked tokens representing vote power for governance, the balance (or locked)
-// balance of the native token (by operator or delegator) should be used.
-// See interchain-security as a reference (although I did not understand some of it),
+// the SDK's gov module and by our oracle module.
 func (k Keeper) IterateBondedValidatorsByPower(
-	sdk.Context, func(int64, stakingtypes.ValidatorI) bool,
+	ctx sdk.Context, f func(int64, stakingtypes.ValidatorI) bool,
 ) {
-	// // we will have at most a 100 validators bonded.
-	// // so it is safe to load all of them up and then call.
-	// validators := k.GetAllExocoreValidators(ctx)
-	// sort.SliceStable(validators, func(i, j int) bool {
-	// 	return validators[i].Power > validators[j].Power
-	// })
-	// for i, v := range validators {
-	// 	pk, err := v.ConsPubKey()
-	// 	if err != nil {
-	// 		// since we stored the validator in the first place, something like this
-	// 		// should never happen, but if it does it is an extremely grave error
-	// 		// that will result in a block mismatch and hence that node will halt.
-	// 		continue
-	// 	}
-	// 	val, err := stakingtypes.NewValidator(nil, pk, stakingtypes.Description{})
-	// 	if err != nil {
-	// 		// same as above.
-	// 		continue
-	// 	}
-
-	// 	// Set validator to bonded status
-	// 	val.Status = stakingtypes.Bonded
-	// 	// Compute tokens from voting power
-	// 	val.Tokens = sdk.TokensFromConsensusPower(v.Power, sdk.DefaultPowerReduction)
-	// 	// #nosec G701 // ok on 64-bit systems.
-	// 	if f(int64(i), val) {
-	// 		break
-	// 	}
-	// }
-	panic("unimplemented on this keeper")
+	prevList := k.GetAllExocoreValidators(ctx)
+	sort.SliceStable(prevList, func(i, j int) bool {
+		return prevList[i].Power > prevList[j].Power
+	})
+	for i, v := range prevList {
+		pk, err := v.ConsPubKey()
+		if err != nil {
+			// will only happen if there is an error in deserialization.
+			continue
+		}
+		val, err := stakingtypes.NewValidator(nil, pk, stakingtypes.Description{})
```

Comment:
`NewValidator` have no power related information(initialized to zero), So if the `f` want the power info, it would cause unexpected behavior

Reply:
As discussed, I have left this for a future pull request and commented out two `x/oracle` unit tests.